common/spring-cloud-stream-modules-common-configuration/src/main/java/org/springframework/cloud/stream/module/file/remote/AbstractRemoteFileSourceProperties.java

```diff
@@ -0,0 +1,161 @@
+
+/**
+ * Common properties for remote file sources (e.g. (S)FTP).
+ *
+ * @author David Turanski
+ * @author Gary Russell
+ *
+ */
+public abstract class AbstractRemoteFileSourceProperties {
+
+	/**
+	 * The remote FTP directory.
+	 */
+	private String remoteDir = "/";
+
+	/**
+	 * Set to true to delete remote files after successful transfer.
+	 */
+	private boolean deleteRemoteFiles = false;
+
+	/**
+	 * The local directory to use for file transfers.
+	 */
+	private File localDir = new File(System.getProperty("java.io.tmpdir") + "/xd/ftp");
+
+	/**
+	 * Set to true to create the local directory if it does not exist.
+	 */
+	private boolean autoCreateLocalDir = true;
+
+	/**
+	 * The suffix to use while the transfer is in progress.
+	 */
+	private String tmpFileSuffix = ".tmp";
+
+	/**
+	 * A filter pattern to match the names of files to transfer.
+	 */
+	private String filenamePattern;
+
+	/**
+	 * A filter regex pattern to match the names of files to transfer.
+	 */
+	private Pattern filenameRegex;
+
+	/**
+	 * The remote file separator.
+	 */
+	private String remoteFileSeparator = "/";
+
+	/**
+	 * Set to true to preserve the original timestamp.
+	 */
+	private boolean preserveTimestamp = true;
+
+	public boolean isAutoCreateLocalDir() {
+		return autoCreateLocalDir;
+	}
+
+	public void setAutoCreateLocalDir(boolean autoCreateLocalDir) {
+		this.autoCreateLocalDir = autoCreateLocalDir;
+	}
+
+	@NotBlank
+	public String getRemoteDir() {
+		return remoteDir;
+	}
+
+	public void setRemoteDir(String remoteDir) {
+		this.remoteDir = remoteDir;
+	}
+
+	public boolean isDeleteRemoteFiles() {
+		return deleteRemoteFiles;
+	}
+
+	public void setDeleteRemoteFiles(boolean deleteRemoteFiles) {
+		this.deleteRemoteFiles = deleteRemoteFiles;
+	}
+
+	@NotNull
+	public File getLocalDir() {
+		return localDir;
+	}
+
+	public void setLocalDir(File localDir) {
+		this.localDir = localDir;
+	}
+
+	@NotBlank
+	public String getTmpFileSuffix() {
+		return tmpFileSuffix;
+	}
+
+	public void setTmpFileSuffix(String tmpFileSuffix) {
+		this.tmpFileSuffix = tmpFileSuffix;
+	}
+
+	public String getFilenamePattern() {
+		return filenamePattern;
+	}
+
+	public void setFilenamePattern(String filenamePattern) {
+		this.filenamePattern = filenamePattern;
+	}
+
+	public Pattern getFilenameRegex() {
+		return filenameRegex;
+	}
+
+	public void setFilenameRegex(Pattern filenameRegex) {
+		this.filenameRegex = filenameRegex;
+	}
+
+	@NotBlank
+	public String getRemoteFileSeparator() {
+		return remoteFileSeparator;
+	}
+
+	public void setRemoteFileSeparator(String remoteFileSeparator) {
+		this.remoteFileSeparator = remoteFileSeparator;
+	}
+
+	public boolean isPreserveTimestamp() {
+		return preserveTimestamp;
+	}
+
+	public void setPreserveTimestamp(boolean preserveTimestamp) {
+		this.preserveTimestamp = preserveTimestamp;
+	}
+
+	@AssertTrue(message = "filenamePattern and filenameRegex are mutually exclusive")
+	public boolean exclusivePatterns() {
```

Comment:
This won't work. Needs to be a javabean style getter (hence `isExclusivePatterns()` or `getExclusivePatterns()`)